### PR TITLE
Add CSS Reset

### LINKED ISSUE #27116: FSE: CSS Reset
In addition to the alignment styles that the front end should provide, we should also provide a CSS reset in Gutenberg to stop each theme having their own reset. Some stuff that is definitely needed:

```
img { height: auto; }
figure { margin: 0; }

```

Please add more as you find it :)

### PR DESCRIPTION
**Pull request by scruffian**
## Description
Adds some CSS reset code to minimize boilerplate code needed to build a theme.

Fixes #27116 

## How has this been tested?
Tested in TT1 blocks

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/style.scss [hunk 1]
```diff
@@ -108,3 +108,14 @@
 .aligncenter {
 	clear: both;
 }
+
+// This resets the WordPress default
+img {
+	height: auto;
```
Comment (aristath) [inline packages/block-library/src/style.scss:114]: Just a note for the `height:auto` here.
In the twentytwentyone theme this was causing a layout shift while the page was loading on slow connections. In order to fix that we had to add inline styles for the featured images (see https://github.com/WordPress/twentytwentyone/pull/692/files).

**Comment by MaggieCabrera** [comment]
https://github.com/WordPress/gutenberg/pull/27517

File: packages/block-library/src/style.scss (hunk 1 above)
Comment (kjellr) [reply packages/block-library/src/style.scss:114]: @aristath Should that fix be baked into core then, and not just into this CSS reset?

If I recall correctly, this rule only became necessary after [lazy loading images launched](https://make.wordpress.org/core/2020/07/14/lazy-loading-images-in-5-5/).

Comment (aristath) [reply packages/block-library/src/style.scss:114]: Yeah, it became an issue after lazy-loading. 
That function does the following:
* Gets the image width & height
* Calculates the aspect-ratio
* Adds `width:100%;height:X%` (X calculated from the aspect-ratio) to the image inline.

If `width:100%` is OK for all themes, then we can definitely port the fix to core, but I don't think all themes will want that...

Comment (kjellr) [reply packages/block-library/src/style.scss:114]: Ah ok, I see what you mean — a 100% width would probably cause its own set of problems in some themes. 😄

But a global fix of some kind seems to be a good idea.

**Comment by scruffian** [comment]
Closing in favour of https://github.com/WordPress/gutenberg/pull/30092

**Pull request closed without merging**